skimage/filters/_gaussian.py

```diff
@@ -113,8 +117,14 @@ def gaussian(image, sigma=1, output=None, mode='nearest', cval=0,
         if len(sigma) != image.ndim:
             sigma = np.concatenate((np.asarray(sigma), [0]))
     image = convert_to_float(image, preserve_range)
-    return ndi.gaussian_filter(image, sigma, mode=mode, cval=cval,
-                               truncate=truncate)
+    if not isinstance(output, np.ndarray):
+        raise ValueError("Provided output argument is not a numpy array")
+    elif image.dtype != output.dtype:
```

Comment:
You also have to check if output is not None here before checking its data type!

Reply:
Added check  if `output` is None for all conditions.